reactor-core/src/main/java/reactor/core/Scannable.java

```diff
@@ -408,19 +418,22 @@ default String name() {
 	 * its chain of {@link #parents()} and {@link #actuals()}.
 	 */
 	default String stepName() {
-		// /!\ this code is duplicated in `InnerConsumer#stepName` in order to use simple class name instead of toString
-
 		/*
 		 * Strip an operator name of various prefixes and suffixes.
 		 * @param name the operator name, usually simpleClassName or fully-qualified classname.
 		 * @return the stripped operator name
 		 */
-		String name = toString();
-		if (name.contains("@") && name.contains("$")) {
-			name = name
-				.substring(0, name.indexOf('$'))
-				.substring(name.lastIndexOf('.') + 1);
+		String name = getClass().getName();
+		int innerClassIndex = name.indexOf('$');
+		if (innerClassIndex != -1) {
+			name = name.substring(0, innerClassIndex);
+		}
+		int stripPackageIndex = name.lastIndexOf('.');
+		if (stripPackageIndex != -1) {
+			name = name.substring(stripPackageIndex + 1);
 		}
+		name = name.substring(name.lastIndexOf('$') + 1);
+		name = name.substring(name.lastIndexOf('.') + 1);
```

Comment:
Is this something you forgot to delete? These have already been covered and can't strip anything more I think.

Reply:
yeah, indeed 👍